### ISSUE TITLE
Nix develop not working with directory that contains special character on path

### ISSUE THREAD

**Issue by arthsmn** [NixOS/nix#9751]
**Describe the bug**
When I'm at a directory that has some accent like 'ó' like `~/Repositórios/ghostty` and try to use `nix develop` it fails.

**Steps To Reproduce**

1. Go to the directory
2. Use `nix develop` or `nix develop .`

**Expected behavior**
It would drop me in a shell to work in the project.

**`nix-env --version` output**
nix-env (Nix) 2.18.1

**Additional context**
Error:
```
fatal: '~/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
warning: could not read HEAD ref from repo at 'file://~/Reposit%FFFFFFC3%FFFFFFB3rios/ghostty', using 'master'
fatal: '~/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
error:
       … while fetching the input 'git+file://~/Reposit%FFFFFFFFFFFFC3%FFFFFFFFFFFFB3rios/ghostty'

       error: program 'git' failed with exit code 128
```

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by thufschmitt** [comment]
This should be fixed on 2.19 (by https://github.com/NixOS/nix/pull/6614 very likely):

```console
$ pwd         
/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty
$ nix run nix/2.18.1 -- build
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
warning: could not read HEAD ref from repo at 'file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFC3%FFFFFFB3rios/ghostty', using 'master'
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
error:
       … while fetching the input 'git+file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFFFFFFFC3%FFFFFFFFFFFFB3rios/ghostty'

       error: program 'git' failed with exit code 128
$ nix run nix/2.19.1 -- build
warning: Git tree '/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty' is dirty
$ 
$ pwd         
/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty
$ nix run nix/2.18.1 -- build
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
warning: could not read HEAD ref from repo at 'file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFC3%FFFFFFB3rios/ghostty', using 'master'
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
error:
       … while fetching the input 'git+file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFFFFFFFC3%FFFFFFFFFFFFB3rios/ghostty'

       error: program 'git' failed with exit code 128
$ nix run nix/2.19.1 -- build
warning: Git tree '/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty' is dirty
$ $ pwd         
/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty
$ nix run nix/2.18.1 -- build
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
warning: could not read HEAD ref from repo at 'file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFC3%FFFFFFB3rios/ghostty', using 'master'
fatal : '/tmp/tmp.NNmSV5bwlR/RepositFFFFC3FFFFB3rios/ghostty' does not appear to be a git repository
fatal : Impossible de lire le dépôt distant.

Veuillez vérifier que vous avez les droits d'accès
et que le dépôt existe.
error:
       … while fetching the input 'git+file:///tmp/tmp.NNmSV5bwlR/Reposit%FFFFFFFFFFFFC3%FFFFFFFFFFFFB3rios/ghostty'

       error: program 'git' failed with exit code 128
$ nix run nix/2.19.1 -- build
warning: Git tree '/tmp/tmp.NNmSV5bwlR/Repositórios/ghostty' is dirty
$  echo $?                                                                            
0
```

**Comment by arthsmn** [comment]
It is now fixed!
